templater/jobs/utils/utils.go

```diff
@@ -46,69 +56,161 @@ func GetJobsByType(repos []string, jobType string) (map[string]map[string]types.
+	if !strings.Contains(fileName, "golang-1-X") {
+		return jobList
+	}
+
+	for _, version := range golangVersions {
+		golangVersionFileName := strings.ReplaceAll(fileName, "golang-1-X", "golang-"+version)
+		goVersion := strings.Replace(version, "-", ".", 1)
+		jobList[golangVersionFileName] = AppendMap(data, map[string]interface{}{
+			"jobGoVersion":  version,
+			"golangVersion": goVersion,
+		})
+	}
+
+	return jobList
+}
+
+func AddPythonVersion(fileName string, data map[string]interface{}) map[string]map[string]interface{} {
+	jobList := map[string]map[string]interface{}{}
+	if !strings.Contains(fileName, "python-3-X") {
+		return jobList
+	}
+
+	for _, version := range pythonVersions {
+		pythonVersionFileName := strings.ReplaceAll(fileName, "python-3-X", "python-"+version)
+		pythonVersion := strings.Replace(version, "-", ".", 1)
+		jobList[pythonVersionFileName] = AppendMap(data, map[string]interface{}{
+			"jobPythonVersion": version,
+			"pythonVersion":    pythonVersion,
+		})
+	}
+
+	return jobList
+}
+
+func AddReleaseBranch(fileName string, data map[string]interface{}) map[string]map[string]interface{} {
+	jobList := map[string]map[string]interface{}{}
+	if !strings.Contains(fileName, "1-X") {
+		return jobList
+	}
+
+	for i, releaseBranch := range releaseBranches {
+		releaseBranchBasedFileName := strings.ReplaceAll(fileName, "1-X", releaseBranch)
+		otherReleaseBranches := append(append([]string{}, releaseBranches[:i]...),
+			releaseBranches[i+1:]...)
+		jobList[releaseBranchBasedFileName] = AppendMap(data, map[string]interface{}{
+			"releaseBranch":        releaseBranch,
+			"otherReleaseBranches": strings.Join(otherReleaseBranches, "|"),
+		})
+
+		// If latest release branch, check if the release branch dir exists before executing cmd
+		// This allows us to experiment with adding prow jobs for new branches without failing other runs
+		if len(releaseBranches)-1 == i {
+			jobList[releaseBranchBasedFileName]["latestReleaseBranch"] = true
+		}
+	}
+
+	return jobList
+}
+
+func RunMappers(jobsToData map[string]map[string]interface{}, mappers []func(string, map[string]interface{}) map[string]map[string]interface{}) {
+	if len(mappers) == 0 {
+		return
+	}
+
+	for fileName, data := range jobsToData {
+		newJobList := mappers[0](fileName, data)
+		if len(newJobList) == 0 {
+			continue
+		}
+
+		for k, v := range newJobList {
+			jobsToData[k] = v
+			if _, ok := data["templateFileName"]; !ok {
+				jobsToData[k]["templateFileName"] = fileName
+			}
+		}
+		delete(jobsToData, fileName)
+
+	}
+
+	RunMappers(jobsToData, mappers[1:])
+}
+
 func UnmarshalJobs(jobDir string) (map[string]types.JobConfig, error) {
-	jobList := map[string]types.JobConfig{}
 
 	files, err := ioutil.ReadDir(jobDir)
 	if err != nil {
 		return nil, fmt.Errorf("error reading job directory %s: %v", jobDir, err)
 	}
 
+	var mappers []func(string, map[string]interface{}) map[string]map[string]interface{}
+	mappers = append(mappers, AddALVersion)
```

Comment:
you can also append multiple -- `mappers = append(mappers, AddALVersion, AddGolangVersion, AddPythonVersion, AddReleaseBranch)`